app/models/attachment.rb

```diff
@@ -27,8 +27,19 @@ def find_or_initialize_by(attributes, &block)
       attributes[:name] = file_digest(file)
       find_by(attributes) || new(attributes.reverse_merge(file_upload: file), &block)
     end
-    # `find_or_create_by(file: file)` is not implemented.
-    private :find_or_create_by
+
+    # Supports `find_or_create_by(file: file)`. Similar to +find_or_initialize_by+, it will try
+    # to return an attachment with the same hash, otherwise, a new attachment is created.
+    #
+    # @param [Hash] attributes The hash attributes with the file.
+    # @return [Attachment] The attachment which contains the file.
+    def find_or_create_by(attributes, &block)
```

Comment:
👍 Split them cause I thought the blocks for `create` and `new` do slightly different things, but I realised it is used to set attributes only. Will fix.